routers/web/repo/issue_view.go

```diff
@@ -40,16 +40,30 @@ import (
 )
 
 // roleDescriptor returns the role descriptor for a comment in/with the given repo, poster and issue
-func roleDescriptor(ctx stdCtx.Context, repo *repo_model.Repository, poster *user_model.User, issue *issues_model.Issue, hasOriginalAuthor bool) (issues_model.RoleDescriptor, error) {
+func roleDescriptor(ctx stdCtx.Context, repo *repo_model.Repository, poster *user_model.User, permsCache map[int64]access_model.Permission, issue *issues_model.Issue, hasOriginalAuthor bool) (issues_model.RoleDescriptor, error) {
 	roleDescriptor := issues_model.RoleDescriptor{}
 
 	if hasOriginalAuthor {
 		return roleDescriptor, nil
 	}
 
-	perm, err := access_model.GetUserRepoPermission(ctx, repo, poster)
-	if err != nil {
-		return roleDescriptor, err
+	var perm access_model.Permission
+	var err error
+	if permsCache != nil {
+		var ok bool
+		perm, ok = permsCache[poster.ID]
+		if !ok {
+			perm, err = access_model.GetUserRepoPermission(ctx, repo, poster)
+			if err != nil {
+				return roleDescriptor, err
+			}
+		}
+		permsCache[poster.ID] = perm
+	} else {
+		perm, err = access_model.GetUserRepoPermission(ctx, repo, poster)
+		if err != nil {
+			return roleDescriptor, err
+		}
```

Comment:
IIRC there is "context cache", but no idea why here it doesn't use it. Forgot, or it's unable to handle this case/ (actually I think the design of "context cache" is not right .....)

Reply:
After trying, I think it's better to keep this code until the context cache can be fixed.